lib/live_cluster/client/node.py

```diff
@@ -530,6 +530,105 @@ async def connect(self, address, port):
     async def refresh_connection(self):
         await self.connect(self.ip, self.port)
 
+    async def needs_refresh(self) -> bool:
+        """
+        Check if node needs refresh based on its service addresses changes.
+        Returns True only if node IP, port, or service addresses have changed.
+        """
+        # If node is not alive, definitely need refresh
+        if not self.alive:
+            logger.debug("Node %s:%s not alive, need refresh", self.ip, self.port)
+            return True
+
+        # If no socket pool, need refresh
+        if not self.socket_pool or self.port not in self.socket_pool:
+            logger.debug("Node %s:%s no socket pool, need refresh", self.ip, self.port)
+            return True
+
+        try:
+            # Get current node info using existing socket
+            temp_sock = await self._get_connection(self.ip, self.port)
+            if not temp_sock:
+                logger.debug("Node %s:%s no socket, need refresh", self.ip, self.port)
+                return True
+
+            # Get current service addresses
+            info_address_call = (
+                self._get_admin_info_call()
+                if self.is_admin_node
+                else self._get_service_info_call()
+            )
```

Comment:
How does this handle cases like the use of the --services-alumni option or --services-alternate ?